src/connection-pool-party.js

```diff
@@ -363,6 +369,20 @@ export default class ConnectionPoolParty extends EventEmitter {
             if (connectNumber > 1) {
               this.reconnectCount += 1;
             }
+            // if all the pools are unhealthy from the last attempt, we
+            // skip trying to re-run the request and go straight to
+            // another heal attempt.
+            // otherwise, we just clear out the unhealthy pools and let
+            // _tryRequest re-identify them.
+            if (!attempts.anyPoolsHealed &&
```

Comment:
when will the next heal attempt occur in this case?

Reply:
it's the next `then` in the chain, [here](https://github.com/godaddy/mssql-pool-party/pull/3/files/e061e02c710d86c45bd9b14e842d4ffbbdd3f6c9#diff-584b3d6075664c775861b1967948f100L389)

Reply:
`return attempts` just jumps to the next `then`, skipping over the request execution

Reply:
so the lifecycle for unhealthy pools is:

1) try heal
2) if any healed -> try making an actual request
3) if none healed or heal fails -> heal again

?

Reply:
yea